Elastica/SchemaGenerator/ContentTypeSchemaGenerator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace OpenOrchestra\Elastica\SchemaGenerator;
+
+use Elastica\Client;
+use OpenOrchestra\Elastica\Mapper\FormToElasticaTypeMapper;
+use OpenOrchestra\ModelInterface\Model\ContentTypeInterface;
+use OpenOrchestra\ModelInterface\Model\FieldTypeInterface;
+
+/**
+ * Class ContentTypeSchemaGenerator
+ */
+class ContentTypeSchemaGenerator implements DocumentToElasticaSchemaGeneratorInterface
+{
+    protected $client;
+    protected $formMapper;
+
+    /**
+     * @param Client                   $client
+     * @param FormToElasticaTypeMapper $formMapper
+     */
+    public function __construct(Client $client, FormToElasticaTypeMapper $formMapper)
+    {
+        $this->client = $client;
+        $this->formMapper = $formMapper;
+    }
+
+    /**
+     * Create a elasticSearch linked to the object
+     *
+     * @param ContentTypeInterface $contentType
+     */
+    public function createMaping($contentType)
+    {
+        $index = $this->client->getIndex('content');
+        $type = $index->getType('content_' . $contentType->getContentTypeId());
+
+        $mapping = array(
```

Comment:
for the static mapping, you can the same mechanic for describe the mapping, with a semantic configuration etc